src/app/collective-rewards/settings/Settings.tsx

```diff
@@ -14,14 +18,34 @@ const settings: Record<SettingType, ReactNode> = {
 }
 
 export const Settings: FC = () => {
+  const { isConnected, address } = useAccount()
   const searchParams = useSearchParams()
-  const settingType = searchParams?.get('type') as SettingType
-
   const router = useRouter()
-
-  if (!isSettingType(settingType)) {
-    router.push('/_not-found')
+  const { data: gauge, isLoading: isLoadingGauge } = useGetBuilderToGauge(address!)
+
+  useEffect(() => {
+    if (!isConnected) {
+      router.replace('/')
+    }
+  }, [isConnected, router])
+
+  const isBuilder = gauge && gauge !== zeroAddress
+  useEffect(() => {
+    if (!isBuilder) {
+      router.replace('/')
+    }
+  }, [gauge, router])
```

Comment:
```suggestion
 }, [gauge, isBuilder, router])
```
If we are going to ignore some dependencies we need to include a comment about it.